src/utils/github/getAuthors.ts

```diff
@@ -26,6 +28,137 @@ export async function getAuthors(
+    return [];
+  }
+}
+
+interface RevertAuthorInfo {
+  login: string | null;
+  email: string | null;
+}
+
+function extractRevertedCommitHash(commitMessage: string): string | null {
+  const match = commitMessage.match(/This reverts commit ([a-f0-9]+)/i);
+  return match ? match[1] : null;
+}
+
+async function getRevertCommitDetails(
+  revertedHash: string,
+  owner: string,
+  repo: string,
+  octokitInstance: Octokit
+): Promise<RevertAuthorInfo | null> {
+  try {
+    const revertedCommitDetailsResponse = await octokitInstance.request(
+      'GET /repos/{owner}/{repo}/commits/{ref}',
+      {
+        owner,
+        repo,
+        ref: revertedHash,
+        headers: { 'X-GitHub-Api-Version': '2022-11-28' },
+      }
+    );
+    const revertedData = revertedCommitDetailsResponse.data;
+
+    const revertedCommitAuthor = revertedData.author
+      ? {
+          login: revertedData.author.login,
+          email: revertedData.commit.author?.email ?? null,
+        }
+      : null;
+
+    return revertedCommitAuthor;
+  } catch (revertError) {
+    console.error(
+      `  Error fetching details for reverted commit ${revertedHash}:`,
+      revertError
+    );
+    return null;
+  }
+}
+
+async function processCommit(
+  commitSha: string,
+  owner: string,
+  repo: string,
+  octokitInstance: Octokit
+): Promise<RevertAuthorInfo | null> {
+  try {
+    const commitDetailsResponse = await octokitInstance.request(
+      'GET /repos/{owner}/{repo}/commits/{ref}',
+      {
+        owner,
+        repo,
+        ref: commitSha,
+        headers: { 'X-GitHub-Api-Version': '2022-11-28' },
+      }
+    );
+    const commitData = commitDetailsResponse.data;
+    const revertedHash = extractRevertedCommitHash(commitData.commit.message);
+    let revertDetails: RevertAuthorInfo | null = null;
+
+    if (revertedHash) {
+      revertDetails = await getRevertCommitDetails(
+        revertedHash,
+        owner,
+        repo,
+        octokitInstance
+      );
+    }
+
+    return revertDetails;
+  } catch (error) {
+    console.error(`Error processing commit ${commitSha}:`, error);
+    return null;
+  }
+}
+
+export async function getAuthorsWithRevertedCommitAuthors(
+  repo: string,
+  baseCommit: string | null,
+  headCommit: string
+): Promise<Array<{ email: string | undefined; login: string | undefined }>> {
+  try {
+    const commitsComparison = await GETSENTRY_ORG.api.repos.compareCommits({
+      owner: GETSENTRY_ORG.slug,
+      repo,
+      base: baseCommit ?? headCommit,
+      head: headCommit,
+    });
+    if (
+      !commitsComparison.data.commits ||
+      commitsComparison.data.commits.length === 0
+    ) {
+      return [];
+    }
+
+    const authors = await Promise.all(
+      commitsComparison.data.commits.map(async (commitMetadata) => {
+        const revertInfo = await processCommit(
+          commitMetadata.sha,
+          GETSENTRY_ORG.slug,
+          repo,
+          GETSENTRY_ORG.api
+        );
```

Comment:
We should avoid making api calls for each commit as we will quickly hit GitHub's rate limits and run into issues. What if we just extract the messages when from the result of the commit comparison above and use those to determine if we need to fetch info about the revert? Also, we maybe actually be able to get the info about the author of the original commit from that info as well and avoid any extra api all together.